internal/resource/route.go

```diff
@@ -94,6 +95,30 @@ func (r Route) Indexes() []model.Index {
 }
 
 func (r Route) Validate(ctx context.Context) error {
+	for _, sni := range r.Route.Snis {
+		if hostnameCheck(sni) != typeName {
+			errWrap := validation.Error{}
+			errWrap.Errs = append(errWrap.Errs, &v1.ErrorDetail{
+				Type:  v1.ErrorType_ERROR_TYPE_FIELD,
+				Field: "snis",
+				Messages: []string{
+					fmt.Sprintf("must not be an IP: '%s'", sni),
+				},
+			})
+			return errWrap
```

Comment:
nothing to do for now: but should we add some utility functions to reduce the boilerplate to build that errWrap.Errs?